Dev1

kamar19 opened the pull request:
Завершение блока задач Андроид 3.

kamar19 committed:
- пункты 1, 2, 3, 4
- Изменение ViewModelListState с состояниями.

kamar19 committed:
загрузка из сети Списка фильмов и жанров

kamar19 committed:
Завершение блока Андроид 3

ermolnik commented on app/src/main/java/com/example/traineemoveapp/compose/filmList/FilmListFragment.kt:
```diff
@@ -26,98 +26,140 @@ import androidx.compose.ui.unit.dp
+) {
+    val state = viewModel.uiFilmsState.collectAsState()
+    val stateGenre = viewModel.uiGenresState.collectAsState()
+    val stateError = viewModel.errorState.collectAsState()
```
Зачем 3 разных Стейта, почему не сделать один на все?

ermolnik commented on app/src/main/java/com/example/traineemoveapp/compose/filmList/FilmListFragment.kt:
```diff
@@ -26,98 +26,140 @@ import androidx.compose.ui.unit.dp
+    }
+
+    when (state.value) {
+        is ViewModelListState.Loading -> {}
```
Зачем остальные стейты, если для них нет никаких вьюшек

ermolnik commented on app/src/main/java/com/example/traineemoveapp/compose/filmList/FilmListItem.kt:
```diff
@@ -20,29 +19,70 @@ import com.example.traineemoveapp.R
+        pathImage = film.posterPicture,
+        descriptionText = film.overview,
+        raiting = film.ratings,
+        pg = film.adult,
```
Что такое pg

ermolnik commented on app/src/main/java/com/example/traineemoveapp/data/Interceptor.kt:
```diff
@@ -0,0 +1,22 @@
+import okhttp3.Interceptor
+import okhttp3.Response
+
+class Interceptor : Interceptor {
```
Класс расширяет сам себя?

kamar19 committed:
- исправления после ревью
- добавление ui для Loading и Error
- исправление бага с пересозданием DetailFilmViewMode
